### DISCUSSION TITLE
Cache CSS error?

### DISCUSSION THREAD

**Discussion by rachel-rigdon-sp** [mkdocs/mkdocs#2826] [Q&A]
Hey! We keep running into a cache error immediately after deploying our docs to WordPress. The CSS and JS are renamed, and the HTML points to the new filenames, but because the browser is cached, it’s still pointing to the old filenames. You can reproduce this error by inspecting the [doc site](https://documentation.sailpoint.com/saas/help/index.html) and removing main.120efc48.min.css

We’ve tried adding cache-control headers, but that hasn’t worked. I know that cache-busting is not supported by mkdocs-material or mkdocs, but has anyone encountered this issue and successfully resolved it?
![skip_to_content](https://user-images.githubusercontent.com/60661988/164291030-bdd69594-37c2-47cd-b6b7-5098d5c78787.png)

**Comment by oprypin** [comment]
I certainly have observed the effect that you are referring to. But I don't fully understand why it happens.
Before I go against your explanation (in the next paragraph), let me just say that I think you're at least very close to the truth and this helps me get closer to understanding.

The explanation you're suggesting sounds strange to me because how is it possible that the browser will cache the HTML itself but not cache the resources? I thought the resources are much more likely to be cached, not the other way around.

And you are referring to the file name such as `main.120efc48.min.css`, also saying

> cache-busting is not supported by mkdocs-material or mkdocs

-well, this is in fact pretty much exactly cache busting, implemented specifically by mkdocs-material.
You would not observe this effect with another theme, they don't normally do this (and mkdocs does nothing in this regard either).
Though then you might observe worse, even if more subtle, effects, because cache busting is still better than no cache busting.
https://github.com/mkdocs/mkdocs/issues/1979#issuecomment-1091683518

Anyway, the lack of cache busting is acknowledged, but this issue is kinda the opposite - it's the cache busting that causes problems somehow. And this *is* unique to mkdocs-material, so maybe you'll get better help asking there.

In summary: I don't know, but I would be happy to find out how this happens.

**Reply by rachel-rigdon-sp** [reply]
Thanks for the response! I've been trying to share what the web guy has been telling me, but you and squidfunk at mkdocs-material have clarified more about what seems to be happening here. DevOps doesn't think it's related to our build and deploy jobs, but we're trying a cache exclusion in WPEngine (not sure why this wasn't tried earlier, but here we are...). Now we're just checking every deployment in the hopes that this doesn't come up again.
